content/articles/ansible-and-nix.md

```diff
@@ -0,0 +1,332 @@
+
+The result of this flaw is that even if you managed to get a server properly
+configured once, you don't know if it will work again.
+If you change your server configuration over time without creating new
+servers, then you need to make sure your configuration files does not diverge
+from the actual state of the server.
+
+Also, even if you tested your playbooks well, it is still possible for them to
+stop working if a system package or application dependencies has been updated
+in your system.
+
+## Nix
+
+This is where the Nix project comes in. It consists of [Nix, The Purely
+Functional Package Manager](https://nixos.org/nix/), [NixOS, The Purely
+Functional Linux Distribution](https://nixos.org/) and [NixOps, The NixOS Cloud
+Deployment Tool](https://nixos.org/nixops/).
+
+Nix produces packages that link to all their dependencies at specific versions
+and that are entirely independent from each other. The result from these
+properties means that you cannot get into the problem illustrated previously. 
+
+NixOS is a simple and modern Linux distribution configured declaratively using
+the same language as Nix packages.
+
+Finally, NixOps is used to configure a «network» of NixOS systems, still using
+the same language.
+
+The learning curve for those tools is steep, the official documentation is
+light and there aren't that many resources available on the web about Nix so I
+had to learn it mostly by reading the source of the tools. Also, NixOps being a
+young project I hit a few open bugs.
+
+The [Nix manual](https://nixos.org/nix/manual/) is a good start to learn about
+the package manager. This [Nix by example
+article](https://medium.com/@MrJamesFisher/nix-by-example-a0063a1a4c55)
+describes the Nix expression language in more detail.
+To understand the rationale behind Nix and NixOS, you can read the accessible
+[NixOS: A Purely Functional Linux
+Distribution](http://nixos.org/~eelco/pubs/nixos-jfp-final.pdf) paper.
+
+For more practical questions, I found that the best way to learn was to find a
+package that resembles what I wanted to do and get inspiration from its
+corresponding expression. NixOS packages can be found using the [Search NixOS
+packages page](https://nixos.org/nixos/packages.html).
+
+
+## Comparing the deployment of a test application
+
+Using the [test
+project](https://github.com/WeAreWizards/voting-deployment-test), I am going to
+illustrate the practical differences between using NixOps and Ansible.
+
+### Dependencies
+
+Nix is first and foremost a package manager, so the most common way to deploy
+applications is to build packages locally and copy them to servers.
+
+Nix being `purely functional`, it should build the same output (a closure)
+given the same input (an expression). To do so, Nix needs all dependencies
+retrieved at build time.
+
+Assuming the dependencies you need are present in the nixos channel and at the
+right version then you can just refer to them like in the main derivation of
```

Comment:
derivation is a new, not yet defined term, needs at least scare quotes or a link